templates/todo/common/auth/js/azureDeveloperCliCredential.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+    AccessToken,
+    TokenCredential,
+} from "@azure/core-auth";
+import { CredentialUnavailableError } from "@azure/identity";
+import child_process from "child_process";
+
+export class AzureDeveloperCliCredential implements TokenCredential {
+    public async getToken(
+        scopes: string | string[],
+    ): Promise<AccessToken> {
+        if (typeof(scopes) === "string") {
+            scopes = [scopes];
+        }
+
+        try {
+            const obj = await getAzdAccessToken(scopes);
+            const isNotLoggedInError = obj.stderr?.match("not logged in, run `azd login` to login");
```

Comment:
Should we support a stronger contract here with `json` outputs?

Reply:
We aren't yet emitting structured JSON for errors, but I think we are in a place now where we could start to do this?  What are you thoughts on what this should look like?

I think we had talked about the general idea being that when run in JSON mode, a command which errors will return a non zero exit code and write some sort of error object to stdout that describes the error.

I know you were thinking about our error story here at some point.  Do you have an idea what we would want this to look like?  Do we want some object with `code` and `message` properties to start, where code is logically a string enum (and this would return something with a type of `NotLoggedIn`?  Map most errors into a generic `GeneralError` type that just has the `String()`'d value of the `error` object we got back?

I can take a cut at some ideas here but if you had input I can just go build out what you were thinking.

Reply:
A `code` and `message` is definitely the starting point, with the ability to map errors into a specific user-facing error code.

I realized that this is simply reformatting the error, so the current change is fine.